### PR TITLE
Leave firewall open

### PR DESCRIPTION
**Pull request by klanclos**
This branch addresses a couple things: one, it implements the firewall_cleanup option that was suggested in Slack. Two, it checks to see whether the firewall hole is already open, and doesn't prompt the user for the password if it's not required. Three, it requires the user to enter a non-empty firewall password when prompted.

### CODE REVIEW

**Review by joshjriley — CHANGES_REQUESTED** [review]
Didn't work for me on my mac.  Debugged it a bit and '-w' is not a valid ping option for mac.  There is '-W' but it is in milliseconds which is different than linux which is seconds.  Took off '-w' option entirely and that actually worked.  Process result was 0 so it returned True.  I saw a '-t' option that looked to match linux option and tried that and it works from command line, but result in code is 64 so it returns false.  Not sure what to suggest for a clean, concise solution.

File: keck_vnc_launcher.py [hunk 1]
```diff
@@ -728,6 +747,56 @@ def close_firewall(self, authpass):
...
+    ## Check to see whether the firewall hole is already open.
+    ##-------------------------------------------------------------------------
+    def test_firewall(self):
+        ''' Return True if the sshuser firewall hole is open; otherwise
+            return False. Also return False if the test cannot be performed.
+        '''
+
+        try:
+            netcat = subprocess.check_output(['which', 'ncat'])
+        except CalledProcessError:
```
Comment (joshjriley) [inline keck_vnc_launcher.py]: Forgot to add that i got crash on name 'CalledProcessError' is not defined.  I changed to 'subprocess.CalledProcessError' and that worked.

Comment (joshjriley) [reply keck_vnc_launcher.py]: btw, the above ping issue didn't cause a crash.  It just didn't skip the firewall auth.

Comment (klanclos) [reply keck_vnc_launcher.py]: Now fixed.

**Comment by klanclos** [comment]
I believe I fixed the ping options to work on OSX as you described. Feel free to adapt what I committed and push whatever change fixes it; should be minor. The intent is to ping once and wait five seconds for a response.

**Comment by klanclos** [comment]
Just closing the loop: Josh R. tested the ping changes, and they worked fine.

**Comment by joshwalawender** [comment]
Merged to dev

**Pull request closed without merging**